authors/bruno-skvorc/polkadot/en_US/final.md

```diff
@@ -0,0 +1,268 @@
+
+## ICMP vs Crosslinks
+
+Crosslinks are how shards in Ethereum will be communicating. By telling the beacon chain that a state change has occured on the shard, the other shards which depend on fresh data from the changed shard need to query it manually. In other words, they need to know what they don't know so that they can request it. This will be a user-land implementation detail and will depend entirely on how wallets, IDEs, protocols and other users of Ethereum implement this communication.
+
+ICMP or inter-chain message passing is Polkadot's way of establishing communication across parachains and parathreads. Each chain and thread have an ingress and egress queue - a queue for incoming and outgoing messages, respectively. Messages are transactions meant for another chain or thread, which are forwarded to the destination parachain or parathread on every block through common collators - collators that are using both of the communicating chains. Since we assume that people will be running collator nodes for different chains at the same time, there should be a reasonably stable mesh effect going on allowing for messages to successfully gossip across to their destination. As a backup, in case some chains or threads are islands and have no other connections, validators will also make sure that ingress and egress queues of the chains and threads they are processing are empty. If not, then they will assume this role of message relayer.
+
+Both modes of communication - crosslinks and ICMP - are asynchronous and thus non-instant. The appearance of real-time communication will be a matter of userland implementation.
+
+## Node types
+
+In Ethereum, the following economy-user and node roles (client software types) exist:
+
+|Name       |Role                                                  |Incentivized|Optional|
+|-----------|------------------------------------------------------|------------|--------|
+|Beacon node|Full node of all shards. Passes on data to validators.|No          |No      |
+|Validator  |Proposes and attests to blocks based on beacon data.  |Yes         |No      |
+|VDF ASIC   |Enhances RANDAO to provide better randomness.         |No          |Yes     |
+
+Beacon nodes can potentially be incentivized by being archive nodes of all shards, which means they keep even the data which was wiped due to state rent expiration. A state-rent wiped contract can be restored by an archive node, for a fee. Archive nodes as-a-service are probable in Ethereum 2.0.
+
+In Polkadot:
+
+|Name       |Role                                                        |Incentivized|Optional |
+|-----------|------------------------------------------------------------|------------|---------|
+|Collator   |Full node of particular shard. Passes on data to validators.|Possibly    |No       |
+|Validator  |Full client of all parachains and parathreads. Proposes and attests to blocks based on collator data.      |Yes         |No       |
```

Comment:
Every validator is not necessarily a full node of all parachains but only has to be to the one it is currently assigned to. A validator proposes relay chain blocks and attests to parachain blocks.